Data/DataRepository.cs

```diff
@@ -497,6 +497,44 @@ public void PostEnrollment(UserCourseEnrollment enrollment)
             }
         }
 
+        public void PostLesson(PostLessonViewModel model)
+        {
+            using (var connection = new MySqlConnection(_connectionString))
+            {
+                int lessonPoints = 0;
+                int userId = 0;
+                connection.Open();
+                foreach (var solution in model.Solutions)
+                {
+                    connection.Execute(
+                        @"INSERT INTO user_solution(component_id, user_id, `status`, points, answer)
```

Comment:
I've already mentioned that if passing lesson again is possible then we need to update existing `user_solution` rather than insert a new one. 

So we need to check, if this lesson is being passed for the first time or not. If not, just update `status`, `points` and `answer` with new values.